akka-remote/src/test/scala/akka/remote/artery/tcp/ssl/PeerSubjectVerifierSpec.scala

```diff
@@ -52,7 +52,7 @@ class PeerSubjectVerifierSpec extends AnyWordSpec with Matchers {
       override def removeValue(name: String): Unit = throw new UnsupportedOperationException()
       override def getValueNames: Array[String] = throw new UnsupportedOperationException()
       override def getLocalCertificates: Array[Certificate] = throw new UnsupportedOperationException()
-      override def getPeerCertificateChain: Array[javax.security.cert.X509Certificate] =
+      override def getPeerCertificateChain/*: Array[javax.security.cert.X509Certificate]*/ =
```

Comment:
I don't think we want these in the PR even if you for some mysterious reason needed them on your machine.

Reply:
before dropping this, do u mind checking if you see this as a warning? if this is the case the it'd be a best practice to clean it up

Reply:
No warning here, on 1.11.0-7, since 1.11.0-8 is not available in the jabba listing yet.

Reply:
hmm, this is actually a scalac thing, not java.

Reply:
my guess, it's related to scalac option `-Xfatal-warnings` which seems to be introduced in AkkaDisciplinePlugin.scala:77 .
this plugin seems to be dominated by a system property `akka.no.discipline`, do you happen to have it defined in your env? does akka's travis/jenkins jobs define it?

Reply:
The akka-remote module is among the "looseProjects" in there so should not have the strict flags on AFAICs. What is the specific warning/failure you get here with the return type in place?

Reply:
Actually 'loose' nowadays just means it doesn't warn about numeric widening, but it's still strict about other things.

I don't see a warning for this line, and I tried a couple of different JVM/Scala version combinations. What JVM and Scala version are you using? What is the exact (fatal) warning you get?

Reply:
this is the warning I get (caret pointing at `: Array`):

> [warn] /Users/efarago/dev/github/akka/akka-remote/src/test/scala/akka/remote/artery/tcp/ssl/PeerSubjectVerifierSpec.scala:55:45: class X509Certificate in package cert is deprecated: see corresponding Javadoc for more information.
[warn]       override def getPeerCertificateChain: Array[javax.security.cert.X509Certificate] =
[warn]                                             ^
[warn] one warning found

re. scala version, thought I'm using 2.13 seems like sbt has a mind of its own:

> akka > show akka-remote/Test/scalaVersion
[info] 2.12.11

re java version:

> [IJ]akka > reload
[info] welcome to sbt 1.3.13 (AdoptOpenJDK Java 11.0.8)
[info] loading global plugins from /Users/efarago/.sbt/1.0/plugins
[info] loading settings for project akka-build from plugins.sbt,idea1.sbt ...
[info] loading project definition from /Users/efarago/dev/github/akka/project
[info] loading settings for project akka from build.sbt ...

strange thing, it now compiles with the return type in place. the reason I modified this in the first place was due to compilation failures on my env.

to summarize: this line does introduce a warning, do you want to include the modification that suppresses it or not?

Reply:
> to summarize: this line does introduce a warning, do you want to include the modification that suppresses it or not?

since this is just the spec I'm OK with keeping the change